solr/contrib/ltr/src/java/org/apache/solr/ltr/LTRScoringQuery.java

```diff
@@ -86,17 +86,15 @@ public LTRScoringQuery(LTRScoringModel ltrScoringModel, boolean extractAllFeatur
 
   public LTRScoringQuery(LTRScoringModel ltrScoringModel, 
       Map<String, String[]> externalFeatureInfo, 
-      boolean extractAllFeatures, LTRThreadModule threadMgr) {
+      boolean extractAllFeatures, LTRThreadModule ltrThreadMgr) {
     this.ltrScoringModel = ltrScoringModel;
     this.efi = externalFeatureInfo;
     this.extractAllFeatures = extractAllFeatures;
-    if (threadMgr != null && threadMgr.getMaxQueryThreads() > 1){
-       ltrThreadMgr = threadMgr;
-       querySemaphore = new Semaphore(ltrThreadMgr.getMaxQueryThreads());
-    }
-    else{
-      ltrThreadMgr = new LTRThreadModule(LTRThreadModule.DEFAULT_MAX_THREADS, LTRThreadModule.DEFAULT_MAX_QUERYTHREADS);
-      querySemaphore = null;
+    this.ltrThreadMgr = ltrThreadMgr;
+    if (this.ltrThreadMgr != null) {
```

Comment:
We should garuntee threadMgr is never null so we dont have to have these null checks